gateway-service-impl/src/main/java/org/hypertrace/gateway/service/explore/ExploreService.java

```diff
@@ -102,6 +109,21 @@ public ExploreResponse explore(RequestContext requestContext, ExploreRequest req
           requestHandler.handleRequest(newExploreRequestContext, request);
 
       return responseBuilder.build();
+    } catch (Exception ex) {
+      boolean acquireSucceeded = rateLimiter.tryAcquire();
+      if (acquireSucceeded || log.isDebugEnabled()) {
```

Comment:
In general, try to avoid embedding error logs and rethrowing. Catch and craft a new error if needed, but always error log in one place - the service entry. Otherwise you're just going to duplicate, because we need to log any thrown errors there anyway.